### PR TITLE
patch for synching civicrm contacts with mailchimp groups

### PR DESCRIPTION
**Pull request by subhashuyadav**

### CODE REVIEW

**Comment by kurund** [comment]
@deepak-srivastava any thoughts on this patch to allow civicrm group to mailchimp groups syncing?

File: CRM/Mailchimp/Sync.php [hunk 1]
```diff
@@ -42,16 +42,16 @@ public function __construct($list_id) {
     $this->list_id = $list_id;
     $this->group_details = CRM_Mailchimp_Utils::getGroupsToSync($groupIDs=[], $list_id, $membership_only=FALSE);
     foreach ($this->group_details as $group_id => $group_details) {
-      if (empty($group_details['category_id'])) {
         $this->membership_group_id = $group_id;
-      }
     }
```
Comment (deepak-srivastava) [inline CRM/Mailchimp/Sync.php]: End of foreach membership_group_id could either point to mailchimp-membership-group or interest-grouping.

File: CRM/Mailchimp/Sync.php [hunk 2]
```diff
@@ -42,16 +42,16 @@ public function __construct($list_id) {
...
     if (empty($this->membership_group_id)) {
       throw new InvalidArgumentException("Failed to find mapped membership group for list '$list_id'");
     }
     // Also cache without the membership group, i.e. interest groups only.
     $this->interest_group_details = $this->group_details;
-    unset($this->interest_group_details[$this->membership_group_id]);
+    if (empty($group_details['category_id'])) {
```
Comment (deepak-srivastava) [inline CRM/Mailchimp/Sync.php]: What i understand: If last group in the list returned by CRM_Mailchimp_Utils::getGroupsToSync() is a membership-group, we unset it from $this->interest_group_details.

File: CRM/Mailchimp/Utils.php [hunk 3]
```diff
@@ -394,10 +394,6 @@ public static function getGroupsToSync($groupIDs = array(), $mc_list_id = null,
       $params[1] = array($mc_list_id, 'String');
     }
 
-    if ($membership_only) {
-      $whereClause .= " AND (mc_grouping_id IS NULL OR mc_grouping_id = '')";
-    }
-
```
Comment (deepak-srivastava) [inline CRM/Mailchimp/Utils.php]: Pull and push sync methods uses this flag to push / pull from mailchimp. 
```
CRM/Mailchimp/Form/Sync.php:    $groups = CRM_Mailchimp_Utils::getGroupsToSync(array(), null, $membership_only = TRUE);
CRM/Mailchimp/Form/Pull.php:      $groups = CRM_Mailchimp_Utils::getGroupsToSync(array(), null, $membership_only=TRUE);

```

Not sure about objective of these changes.

**Comment by deepak-srivastava** [comment]
@subhashuyadav Objective of patch isn't quite clear. Could you clarify please?

**Comment by deepak-srivastava** [comment]
Closing for no response. Please re-open if you want to discuss.

**Pull request closed without merging**